skr-api/README.md

```diff
@@ -0,0 +1,62 @@
+# Secure Key Relase (SKR) at Runtime 
+
+## Architecture
+
+SKR is deployed as 2 components:
+
+### attester
+
+A privileged process, interacting with TEE platform services and devices. It exposes a simple API to gather TEE evidence via   Unix Domain Socket. It is packaged with the podvm image and runs as a systemd unit. `agent-protocol-forwarder` will inject the process' socket into each container as a mount entry.
+
+### skr-api
+
+A process that can be deployed as a sidecar in a pod, it'll access the `attester` socket to retrieve TEE evidence from the podvm in a remote attestation flow with an external KBS/verifier. It exposes a simple http API on `localhost:50080` for other pods to use.
+
+## Build
+
+The attester component is built as part of the podvm image build. skr-api can be via Docker:
+
+```bash
+export $my_kbs="https://mykbs.com"
+export $my_image="example.com/image:tag"
+docker build -t "$my_image" .
+docker push "$my_image" 
+cat << EOF > nginx-caa.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nginx-caa
+  name: nginx-caa
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx-caa
+  template:
+    metadata:
+      labels:
+        app: nginx-caa
+    spec:
+      runtimeClassName: kata-remote
+      containers:
+      - image: nginx:stable
+        name: nginx
+      - image: $my_image
+        name: skr-api
+        command: ["/skr-api", "-k", "$my_kbs" ]
```

Comment:
I wonder if we can have a volume that does host mount of the `/run/confidential-containers/attester.sock`. And not include any of the changes that are done in the interceptor.

```suggestion
        command: ["/skr-api", "-k", "$my_kbs" ]
        volumeMounts:
        - mountPath: /run/confidential-containers/attester.sock
          name: socket-volume
      volumes:
      - name: socket-volume
        hostPath:
          # directory location on host
          path: /run/confidential-containers/attester.sock
          type: Socket
```

Reply:
that would attempt to mount a socket on the k8s node (the "host"), not on the podvm, right?

Reply:
right

Reply:
Yep makes sense.